src/Android/Xamarin.Android/Samples/Analysis/LineOfSightGeoElement/LineOfSightGeoElement.cs

```diff
@@ -0,0 +1,286 @@
+// language governing permissions and limitations under the License.
+
+using Android.App;
+using Android.OS;
+using Android.Widget;
+using ArcGISRuntimeXamarin.Managers;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.UI;
+using Esri.ArcGISRuntime.UI.Controls;
+using Esri.ArcGISRuntime.UI.GeoAnalysis;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace ArcGISRuntimeXamarin.Samples.LineOfSightGeoElement
+{
+    [Activity]
+    public class LineOfSightGeoElement : Activity
+    {
+        // Create and hold the SceneView
+        private SceneView _mySceneView = new SceneView();
+
+        // Hold the label that will show the analysis status
+        private TextView _myStatusLabel;
+
+        // Hold the slider that will control observer height
+        private SeekBar _myHeightSlider;
+
+        // URL of the elevation service - provides elevation component of the scene
+        private Uri _elevationUri = new Uri("http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer");
+
+        // URL of the building service - provides builidng models
+        private Uri _buildingsUri = new Uri("https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/New_York_LoD2_3D_Buildings/SceneServer/layers/0");
+
+        // Starting point of the observation point
+        private MapPoint _observerPoint = new MapPoint(-73.984988, 40.748131, 20, SpatialReferences.Wgs84);
+
+        // Graphic to represent the observation point
+        private Graphic observerGraphic;
+
+        // Graphic to represent the observed target
+        private Graphic _taxiGraphic;
+
+        // Line of Sight Analysis
+        private GeoElementLineOfSight _geoLine;
+
+        // For taxi animation - four points in a loop
+        private MapPoint[] points = { new MapPoint(-73.984513, 40.748469, SpatialReferences.Wgs84),
+                                      new MapPoint(-73.985068, 40.747786, SpatialReferences.Wgs84),
+                                      new MapPoint(-73.983452, 40.747091, SpatialReferences.Wgs84),
+                                      new MapPoint(-73.982961, 40.747762, SpatialReferences.Wgs84)
+        };
+
+        // For taxi animation - tracks animation state
+        private int pointIndex = 0;
+        private int frameIndex = 0;
+        private int frameMax = 150;
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Line of Sight (GeoElement)";
+
+            // Create the UI, setup the control references and execute initialization
+            CreateLayout();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Create scene
+            Scene myScene = new Scene(Basemap.CreateImageryWithLabels());
+            // 1. Set initial viewpoint
```

Comment:
Do we need the lines of code comments to be numbered? It seems clear that the lines come on after another. The white space between blocks of codes is enough to distinguish when creating the scene has been completed and we start to add the observer.

Reply:
The intention was to separate out the logical groups of actions. There are several multi-step processes involved with initializing the sample and it felt like soup without that.